hypertrace-core-graphql-span-schema/src/main/java/org/hypertrace/core/graphql/span/dao/SpanLogEventRequestBuilder.java

```diff
@@ -29,31 +33,67 @@
+    this.attributeStore = attributeStore;
   }
 
   Single<LogEventsRequest> buildLogEventsRequest(
       SpanRequest gqlRequest, SpansResponse spansResponse) {
-    return zip(
-        this.attributeConverter.convert(gqlRequest.logEventAttributes()),
-        buildLogEventsQueryFilter(gqlRequest, spansResponse).flatMap(filterConverter::convert),
-        (selections, filter) ->
-            LogEventsRequest.newBuilder()
-                .setStartTimeMillis(
-                    gqlRequest.spanEventsRequest().timeRange().startTime().toEpochMilli())
-                .setEndTimeMillis(
-                    gqlRequest.spanEventsRequest().timeRange().endTime().toEpochMilli())
-                .addAllSelection(selections)
-                .setFilter(filter)
-                .build());
+    return getRequestAttributes(
+            gqlRequest.spanEventsRequest().context(), gqlRequest.logEventAttributes())
+        .flatMap(
+            v ->
+                zip(
+                    this.attributeConverter.convert(v),
+                    buildLogEventsQueryFilter(gqlRequest, spansResponse)
+                        .flatMap(filterConverter::convert),
+                    (selections, filter) ->
+                        LogEventsRequest.newBuilder()
+                            .setStartTimeMillis(
+                                gqlRequest
+                                    .spanEventsRequest()
+                                    .timeRange()
+                                    .startTime()
+                                    .toEpochMilli())
+                            .setEndTimeMillis(
+                                gqlRequest.spanEventsRequest().timeRange().endTime().toEpochMilli())
+                            .addAllSelection(selections)
+                            .setFilter(filter)
+                            .build()));
+  }
+
+  private Single<Collection<AttributeRequest>> getRequestAttributes(
+      GraphQlRequestContext requestContext, Collection<AttributeRequest> logEventAttributes) {
+    return this.attributeStore
+        .getForeignIdAttribute(
+            requestContext,
+            HypertraceCoreAttributeScopeString.LOG_EVENT,
+            HypertraceCoreAttributeScopeString.SPAN)
+        .flatMap(
+            spanId ->
+                logEventAttributes.stream()
+                        .anyMatch(
+                            logEventAttribute ->
+                                logEventAttribute.attribute().key().equals(spanId.key()))
+                    ? Single.just(logEventAttributes)
+                    : updateRequestAttribute(spanId, logEventAttributes));
+  }
+
+  private Single<Collection<AttributeRequest>> updateRequestAttribute(
+      AttributeModel attributeModel, Collection<AttributeRequest> requests) {
+    List<AttributeRequest> list = new ArrayList<>(requests);
+    list.add(new DefaultAttributeRequest(attributeModel, ""));
+    return Single.just(list);
```

Comment:
I am guessing this is a mutable list. Can we convert it into an unmodifiable list? Basically, you can use `Collections.unmodifiableList(list)`

Reply:
This method is removed now (as per Aaron's suggestion).